diff -Nru util-linux-ng-2.16.1-orig/config.h util-linux-ng-2.16.1/config.h
--- util-linux-ng-2.16.1-orig/config.h	1970-01-01 01:00:00.000000000 +0100
+++ util-linux-ng-2.16.1/config.h	2009-09-30 15:52:11.000000000 +0200
@@ -0,0 +1,469 @@
+/* config.h.in.  Generated from configure.ac by autoheader.  */
+
+/* Define if building universal (internal helper macro) */
+#undef AC_APPLE_UNIVERSAL_BUILD
+
+/* Define to 1 if translation of program messages to the user's native
+   language is requested. */
+#undef ENABLE_NLS
+
+/* Define to 1 if you have the <blkid/blkid.h> header file. */
+#undef HAVE_BLKID_BLKID_H
+
+/* Define to 1 if you have the blkid_evaluate_tag(). */
+#undef HAVE_BLKID_EVALUATE_TAG
+
+/* Define to 1 if you have the <blkid.h> header file. */
+#undef HAVE_BLKID_H
+
+/* Define if the GNU dcgettext() function is already present or preinstalled.
+   */
+#undef HAVE_DCGETTEXT
+
+/* Define to 1 if you have the declaration of `ADDR_COMPAT_LAYOUT', and to 0
+   if you don't. */
+#undef HAVE_DECL_ADDR_COMPAT_LAYOUT
+
+/* Define to 1 if you have the declaration of `ADDR_LIMIT_32BIT', and to 0 if
+   you don't. */
+#undef HAVE_DECL_ADDR_LIMIT_32BIT
+
+/* Define to 1 if you have the declaration of `ADDR_LIMIT_3GB', and to 0 if
+   you don't. */
+#undef HAVE_DECL_ADDR_LIMIT_3GB
+
+/* Define to 1 if you have the declaration of `ADDR_NO_RANDOMIZE', and to 0 if
+   you don't. */
+#undef HAVE_DECL_ADDR_NO_RANDOMIZE
+
+/* Define to 1 if you have the declaration of `FDPIC_FUNCPTRS', and to 0 if
+   you don't. */
+#undef HAVE_DECL_FDPIC_FUNCPTRS
+
+/* Define to 1 if you have the declaration of `MMAP_PAGE_ZERO', and to 0 if
+   you don't. */
+#undef HAVE_DECL_MMAP_PAGE_ZERO
+
+/* Define to 1 if you have the declaration of `READ_IMPLIES_EXEC', and to 0 if
+   you don't. */
+#undef HAVE_DECL_READ_IMPLIES_EXEC
+
+/* Define to 1 if you have the declaration of `STICKY_TIMEOUTS', and to 0 if
+   you don't. */
+#undef HAVE_DECL_STICKY_TIMEOUTS
+
+/* Define to 1 if you have the declaration of `WHOLE_SECONDS', and to 0 if you
+   don't. */
+#undef HAVE_DECL_WHOLE_SECONDS
+
+/* Define to 1 if you have the declaration of `_NL_TIME_WEEK_1STDAY', and to 0
+   if you don't. */
+#undef HAVE_DECL__NL_TIME_WEEK_1STDAY
+
+/* Define to 1 if you have the <dlfcn.h> header file. */
+#undef HAVE_DLFCN_H
+
+/* Define to 1 if you have the <errno.h> header file. */
+#define HAVE_ERRNO_H 1
+
+/* Define to 1 if you have the <err.h> header file. */
+#undef HAVE_ERR_H
+
+/* Define to 1 if fseeko (and presumably ftello) exists and is declared. */
+#define HAVE_FSEEKO 1
+
+/* Define to 1 if you have the `fstatat' function. */
+#undef HAVE_FSTATAT
+
+/* Define to 1 if you have the `fsync' function. */
+#define HAVE_FSYNC 1
+
+/* Define to 1 if you have the `getdomainname' function. */
+#undef HAVE_GETDOMAINNAME
+
+/* Define to 1 if you have the `getdtablesize' function. */
+#define HAVE_GETDTABLESIZE 1
+
+/* Define to 1 if you have the <getopt.h> header file. */
+#define HAVE_GETOPT_H 1
+
+/* Define to 1 if you have the `getrlimit' function. */
+#undef HAVE_GETRLIMIT
+
+/* Define if the GNU gettext() function is already present or preinstalled. */
+#undef HAVE_GETTEXT
+
+/* Define to 1 if you have the `get_current_dir_name' function. */
+#undef HAVE_GET_CURRENT_DIR_NAME
+
+/* Define if you have the iconv() function. */
+#undef HAVE_ICONV
+
+/* Define to 1 if you have the `inet_aton' function. */
+#undef HAVE_INET_ATON
+
+/* Define to 1 if you have the `inotify_init' function. */
+#undef HAVE_INOTIFY_INIT
+
+/* Define to 1 if you have the <inttypes.h> header file. */
+#define HAVE_INTTYPES_H 1
+
+/* Define to 1 if you have the `jrand48' function. */
+#undef HAVE_JRAND48
+
+/* Define to 1 if you have the <langinfo.h> header file. */
+#undef HAVE_LANGINFO_H
+
+/* Define to 1 if you have the `lchown' function. */
+#undef HAVE_LCHOWN
+
+/* Define to 1 if you have the `audit' library (-laudit). */
+#undef HAVE_LIBAUDIT
+
+/* Define to 1 if you have the -lblkid. */
+#undef HAVE_LIBBLKID
+
+/* Define to 1 if you have the `ncurses' library (-lncurses). */
+#define HAVE_LIBNCURSES 1
+
+/* Define to 1 if you have the `ncursesw' library (-lncursesw). */
+#undef HAVE_LIBNCURSESW
+
+/* Define to 1 if you have the `selinux' library (-lselinux). */
+#undef HAVE_LIBSELINUX
+
+/* Define to 1 if you have the `termcap' library (-ltermcap). */
+#define HAVE_LIBTERMCAP 1
+
+/* Define to 1 if you have the `util' library (-lutil). */
+#undef HAVE_LIBUTIL
+
+/* Define to 1 if you have the -luuid. */
+#undef HAVE_LIBUUID
+
+/* Define to 1 if you have the `z' library (-lz). */
+#define HAVE_LIBZ 1
+
+/* Define to 1 if you have the <linux/blkpg.h> header file. */
+#undef HAVE_LINUX_BLKPG_H
+
+/* Define to 1 if you have the <linux/compiler.h> header file. */
+#undef HAVE_LINUX_COMPILER_H
+
+/* Define to 1 if you have the <linux/fd.h> header file. */
+#undef HAVE_LINUX_FD_H
+
+/* Define to 1 if you have the <linux/major.h> header file. */
+#undef HAVE_LINUX_MAJOR_H
+
+/* Define to 1 if you have the <linux/raw.h> header file. */
+#undef HAVE_LINUX_RAW_H
+
+/* Define to 1 if you have the <linux/tiocl.h> header file. */
+#undef HAVE_LINUX_TIOCL_H
+
+/* Define to 1 if you have the <linux/version.h> header file. */
+#undef HAVE_LINUX_VERSION_H
+
+/* Define to 1 if you have the `llseek' function. */
+#undef HAVE_LLSEEK
+
+/* Define to 1 if you have the <locale.h> header file. */
+#define HAVE_LOCALE_H 1
+
+/* Define to 1 if you have the `lseek64' function. */
+#undef HAVE_LSEEK64
+
+/* Define to 1 if you have the <memory.h> header file. */
+#undef HAVE_MEMORY_H
+
+/* Define to 1 if you have the <mntent.h> header file. */
+#undef HAVE_MNTENT_H
+
+/* Define to 1 if you have the `nanosleep' function. */
+#define HAVE_NANOSLEEP 1
+
+/* Define to 1 if you have the <ncurses.h> header file. */
+#define HAVE_NCURSES_H 1
+
+/* Define to 1 if you have the <ncurses/ncurses.h> header file. */
+#undef HAVE_NCURSES_NCURSES_H
+
+/* Define to 1 if you have the <netinet/in.h> header file. */
+#undef HAVE_NETINET_IN_H
+
+/* Define to 1 if you have the <net/if_dl.h> header file. */
+#undef HAVE_NET_IF_DL_H
+
+/* Define to 1 if you have the <net/if.h> header file. */
+#undef HAVE_NET_IF_H
+
+/* Define to 1 if you have the `openat' function. */
+#undef HAVE_OPENAT
+
+/* Define to 1 if you have the `personality' function. */
+#undef HAVE_PERSONALITY
+
+/* Define to 1 if you have the `prctl' function. */
+#undef HAVE_PRCTL
+
+/* Define to 1 if you have the <pty.h> header file. */
+#undef HAVE_PTY_H
+
+/* Define to 1 if you have the <rpcsvc/nfs_prot.h> header file. */
+#undef HAVE_RPCSVC_NFS_PROT_H
+
+/* Define to 1 if you have the `rpmatch' function. */
+#undef HAVE_RPMATCH
+
+/* Define if struct sockaddr contains sa_len */
+#undef HAVE_SA_LEN
+
+/* Define to 1 if you have the <scsi/scsi.h> header file. */
+#undef HAVE_SCSI_SCSI_H
+
+/* Define to 1 if you have the `security_get_initial_context' function. */
+#undef HAVE_SECURITY_GET_INITIAL_CONTEXT
+
+/* Define to 1 if you have the <security/pam_misc.h> header file. */
+#undef HAVE_SECURITY_PAM_MISC_H
+
+/* Define to 1 if you have the `setresgid' function. */
+#undef HAVE_SETRESGID
+
+/* Define to 1 if you have the `setresuid' function. */
+#undef HAVE_SETRESUID
+
+/* Define to 1 if you have the <slang/slcurses.h> header file. */
+#undef HAVE_SLANG_SLCURSES_H
+
+/* Define to 1 if you have the <slcurses.h> header file. */
+#undef HAVE_SLCURSES_H
+
+/* Define to 1 if you have the `srandom' function. */
+#undef HAVE_SRANDOM
+
+/* Define to 1 if you have the <stdint.h> header file. */
+#define HAVE_STDINT_H 1
+
+/* Define to 1 if you have the <stdlib.h> header file. */
+#define HAVE_STDLIB_H 1
+
+/* Define to 1 if you have the <strings.h> header file. */
+#undef HAVE_STRINGS_H
+
+/* Define to 1 if you have the <string.h> header file. */
+#define HAVE_STRING_H 1
+
+/* Define to 1 if you have the `strtoull' function. */
+#define HAVE_STRTOULL 1
+
+/* Define to 1 if `c_line' is member of `struct termios'. */
+#undef HAVE_STRUCT_TERMIOS_C_LINE
+
+/* Define to 1 if you have the `sysconf' function. */
+#undef HAVE_SYSCONF
+
+/* Define to 1 if you have the <sys/disklabel.h> header file. */
+#undef HAVE_SYS_DISKLABEL_H
+
+/* Define to 1 if you have the <sys/disk.h> header file. */
+#undef HAVE_SYS_DISK_H
+
+/* Define to 1 if you have the <sys/file.h> header file. */
+#undef HAVE_SYS_FILE_H
+
+/* Define to 1 if you have the <sys/ioctl.h> header file. */
+#define HAVE_SYS_IOCTL_H 1
+
+/* Define to 1 if you have the <sys/io.h> header file. */
+#undef HAVE_SYS_IO_H
+
+/* Define to 1 if you have the <sys/mkdev.h> header file. */
+#undef HAVE_SYS_MKDEV_H
+
+/* Define to 1 if you have the <sys/prctl.h> header file. */
+#undef HAVE_SYS_PRCTL_H
+
+/* Define to 1 if you have the <sys/queue.h> header file. */
+#undef HAVE_SYS_QUEUE_H
+
+/* Define to 1 if you have the <sys/resource.h> header file. */
+#undef HAVE_SYS_RESOURCE_H
+
+/* Define to 1 if you have the <sys/sockio.h> header file. */
+#undef HAVE_SYS_SOCKIO_H
+
+/* Define to 1 if you have the <sys/stat.h> header file. */
+#define HAVE_SYS_STAT_H 1
+
+/* Define to 1 if you have the <sys/swap.h> header file. */
+#undef HAVE_SYS_SWAP_H
+
+/* Define to 1 if you have the <sys/time.h> header file. */
+#define HAVE_SYS_TIME_H 1
+
+/* Define to 1 if you have the <sys/types.h> header file. */
+#define HAVE_SYS_TYPES_H 1
+
+/* Define to 1 if you have the <sys/un.h> header file. */
+#undef HAVE_SYS_UN_H
+
+/* Define to 1 if you have the <sys/user.h> header file. */
+#undef HAVE_SYS_USER_H
+
+/* Does struct tm have a field tm_gmtoff? */
+#undef HAVE_TM_GMTOFF
+
+/* Define to 1 if you have the <unistd.h> header file. */
+#define HAVE_UNISTD_H 1
+
+/* Define to 1 if you have the `unlinkat' function. */
+#undef HAVE_UNLINKAT
+
+/* Define to 1 if you have the `updwtmp' function. */
+#undef HAVE_UPDWTMP
+
+/* Define to 1 if you want to use uuid daemon. */
+#undef HAVE_UUIDD
+
+/* Define to 1 if you have the <uuid.h> header file. */
+#undef HAVE_UUID_H
+
+/* Define to 1 if you have the <uuid/uuid.h> header file. */
+#undef HAVE_UUID_UUID_H
+
+/* Do we have wide character support? */
+#undef HAVE_WIDECHAR
+
+/* Do we have __progname? */
+#undef HAVE___PROGNAME
+
+/* Define to 1 if you have the `__secure_getenv' function. */
+#undef HAVE___SECURE_GETENV
+
+/* libblkid date string */
+#undef LIBBLKID_DATE
+
+/* libblkid version string */
+#undef LIBBLKID_VERSION
+
+/* Should login chown /dev/vcsN? */
+#undef LOGIN_CHOWN_VCS
+
+/* Should login stat() the mailbox? */
+#undef LOGIN_STAT_MAIL
+
+/* Define to the sub-directory in which libtool stores uninstalled libraries.
+   */
+#undef LT_OBJDIR
+
+/* Do we need -lcrypt? */
+#undef NEED_LIBCRYPT
+
+/* Should chsh allow only shells in /etc/shells? */
+#undef ONLY_LISTED_SHELLS
+
+/* Name of package */
+#undef PACKAGE
+
+/* Define to the address where bug reports for this package should be sent. */
+#undef PACKAGE_BUGREPORT
+
+/* Define to the full name of this package. */
+#undef PACKAGE_NAME
+
+/* Define to the full name and version of this package. */
+#undef PACKAGE_STRING
+
+/* Define to the one symbol short name of this package. */
+#undef PACKAGE_TARNAME
+
+/* Define to the version of this package. */
+#undef PACKAGE_VERSION
+
+/* Should pg ring the bell on invalid keys? */
+#undef PG_BELL
+
+/* Should chfn and chsh require the user to enter the password? */
+#undef REQUIRE_PASSWORD
+
+/* Define to 1 if you have the ANSI C header files. */
+#undef STDC_HEADERS
+
+/* Is swapon() declared with two parameters? */
+#undef SWAPON_HAS_TWO_ARGS
+
+/* Fallback syscall number for ioprio_get */
+#undef SYS_ioprio_get
+
+/* Fallback syscall number for ioprio_set */
+#undef SYS_ioprio_set
+
+/* Fallback syscall number for pivot_root */
+#undef SYS_pivot_root
+
+/* Fallback syscall number for sched_getaffinity */
+#undef SYS_sched_getaffinity
+
+/* If the compiler supports a TLS storage class define it to that here */
+#undef TLS
+
+/* Enable extensions on AIX 3, Interix.  */
+#ifndef _ALL_SOURCE
+# undef _ALL_SOURCE
+#endif
+/* Enable GNU extensions on systems that have them.  */
+#ifndef _GNU_SOURCE
+# undef _GNU_SOURCE
+#endif
+/* Enable threading extensions on Solaris.  */
+#ifndef _POSIX_PTHREAD_SEMANTICS
+# undef _POSIX_PTHREAD_SEMANTICS
+#endif
+/* Enable extensions on HP NonStop.  */
+#ifndef _TANDEM_SOURCE
+# undef _TANDEM_SOURCE
+#endif
+/* Enable general extensions on Solaris.  */
+#ifndef __EXTENSIONS__
+# undef __EXTENSIONS__
+#endif
+
+
+/* Should wall and write be installed setgid tty? */
+#undef USE_TTY_GROUP
+
+/* Version number of package */
+#undef VERSION
+
+/* Define WORDS_BIGENDIAN to 1 if your processor stores words with the most
+   significant byte first (like Motorola and SPARC, unlike Intel). */
+#if defined AC_APPLE_UNIVERSAL_BUILD
+# if defined __BIG_ENDIAN__
+#  define WORDS_BIGENDIAN 1
+# endif
+#else
+# ifndef WORDS_BIGENDIAN
+#  undef WORDS_BIGENDIAN
+# endif
+#endif
+
+/* Number of bits in a file offset, on hosts where this is settable. */
+#undef _FILE_OFFSET_BITS
+
+/* Define to 1 to make fseeko visible on some hosts (e.g. glibc 2.2). */
+#undef _LARGEFILE_SOURCE
+
+/* Define for large files, on AIX-style hosts. */
+#undef _LARGE_FILES
+
+/* Define to 1 if on MINIX. */
+#undef _MINIX
+
+/* Define to 2 if the system does not provide POSIX.1 features except with
+   this defined. */
+#undef _POSIX_1_SOURCE
+
+/* Define to 1 if you need to in order for `stat' and other things to work. */
+#undef _POSIX_SOURCE
diff -Nru util-linux-ng-2.16.1-orig/fdisk/cfdisk.c util-linux-ng-2.16.1/fdisk/cfdisk.c
--- util-linux-ng-2.16.1-orig/fdisk/cfdisk.c	2009-09-07 09:24:49.000000000 +0200
+++ util-linux-ng-2.16.1/fdisk/cfdisk.c	2009-09-30 16:01:41.000000000 +0200
@@ -56,6 +56,10 @@
  *
  ****************************************************************************/
 
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif /* HAVE_CONFIG_H */
+
 #include <stdlib.h>
 #include <stdio.h>
 #include <stdarg.h>
@@ -1685,7 +1689,7 @@
 	 clear_warning();
     }
 
-#ifdef BLKFLSBUF
+#if 0
     /* Blocks are visible in more than one way:
        e.g. as block on /dev/hda and as block on /dev/hda3
        By a bug in the Linux buffer cache, we will see the old
@@ -1907,7 +1911,7 @@
 	}
 
     if (is_bdev) {
-#ifdef BLKRRPART
+#if 0
 	 sync();
 	 sleep(2);
 	 if (!ioctl(fd,BLKRRPART))
@@ -2655,7 +2659,7 @@
 	mvaddstr(WARNING_START, 0, line);
 
 
-    snprintf(line, COLS+1, "cfdisk (%s)", PACKAGE_STRING);
+    snprintf(line, COLS+1, "cfdisk");
     mvaddstr(HEADER_START, (COLS-strlen(line))/2, line);
     snprintf(line, COLS+1, _("Disk Drive: %s"), disk_device);
     mvaddstr(HEADER_START+2, (COLS-strlen(line))/2, line);
@@ -2972,7 +2976,7 @@
 	    }
 	    break;
 	case 'v':
-	    fprintf(stderr, "cfdisk (%s)\n", PACKAGE_STRING);
+	    fprintf(stderr, "cfdisk\n");
 	    copyright();
 	    exit(0);
 	case 'z':
diff -Nru util-linux-ng-2.16.1-orig/fdisk/gpt.c util-linux-ng-2.16.1/fdisk/gpt.c
--- util-linux-ng-2.16.1-orig/fdisk/gpt.c	2009-07-04 01:20:02.000000000 +0200
+++ util-linux-ng-2.16.1/fdisk/gpt.c	2009-09-30 16:02:26.000000000 +0200
@@ -31,7 +31,6 @@
 #include <inttypes.h>
 #include <sys/stat.h>
 #include <sys/ioctl.h>
-#include <sys/utsname.h>
 #include <sys/types.h>
 #include <fcntl.h>
 #include <unistd.h>
diff -Nru util-linux-ng-2.16.1-orig/lib/blkdev.c util-linux-ng-2.16.1/lib/blkdev.c
--- util-linux-ng-2.16.1-orig/lib/blkdev.c	2009-07-04 01:20:00.000000000 +0200
+++ util-linux-ng-2.16.1/lib/blkdev.c	2009-09-30 16:06:11.000000000 +0200
@@ -41,33 +41,17 @@
 int
 blkdev_get_size(int fd, unsigned long long *bytes)
 {
-	/* TODO: use stat as well */
+	int error;
+	device_geometry_t geometry;
 
-#ifdef BLKGETSIZE64
-#ifdef __linux__
-	int ver = get_linux_version();
-	/* kernels 2.4.15-2.4.17, had a broken BLKGETSIZE64 */
-	if (ver >= KERNEL_VERSION (2,6,0) ||
-	   (ver >= KERNEL_VERSION (2,4,18) && ver < KERNEL_VERSION (2,5,0)))
-#endif
-		if (ioctl(fd, BLKGETSIZE64, bytes) >= 0)
-			return 0;
-#endif /* BLKGETSIZE64 */
-
-#ifdef BLKGETSIZE
-	{
-		unsigned long size;
-
-		if (ioctl(fd, BLKGETSIZE, &size) >= 0) {
-			*bytes = ((unsigned long long)size << 9);
-			return 0;
-		}
+	error = ioctl( fd, IOCTL_DISK_GET_GEOMETRY, &geometry );
+
+	if ( error < 0 ) {
+		return -1;
 	}
 
-	return -1;
-#endif /* BLKGETSIZE */
+	*bytes = geometry.sector_count * geometry.bytes_per_sector;
 
-	*bytes = blkdev_find_size(fd);
 	return 0;
 }
 
@@ -75,35 +59,36 @@
 int
 blkdev_get_sectors(int fd, unsigned long long *sectors)
 {
-	unsigned long long bytes;
+	int error;
+	device_geometry_t geometry;
+
+	error = ioctl( fd, IOCTL_DISK_GET_GEOMETRY, &geometry );
 
-	if (blkdev_get_size(fd, &bytes) == 0) {
-		*sectors = (bytes >> 9);
-		return 0;
+	if ( error < 0 ) {
+		return -1;
 	}
 
-	return -1;
+	*sectors = geometry.sector_count;
+
+	return 0;
 }
 
 /* get hardware sector size */
 int
 blkdev_get_sector_size(int fd, int *sector_size)
 {
-#ifdef BLKSSZGET
-#ifdef __linux__
-	if (get_linux_version() < KERNEL_VERSION(2,3,3)) {
-		*sector_size = DEFAULT_SECTOR_SIZE;
-		return 0;
+	int error;
+	device_geometry_t geometry;
+
+	error = ioctl( fd, IOCTL_DISK_GET_GEOMETRY, &geometry );
+
+	if ( error < 0 ) {
+		return -1;
 	}
-#endif
-	if (ioctl(fd, BLKSSZGET, sector_size) >= 0)
-		return 0;
-
-	return -1;
-#else
-	*sector_size = DEFAULT_SECTOR_SIZE;
+
+	*sector_size = geometry.bytes_per_sector;
+
 	return 0;
-#endif
 }
 
 
diff -Nru util-linux-ng-2.16.1-orig/pbuild.xml util-linux-ng-2.16.1/pbuild.xml
--- util-linux-ng-2.16.1-orig/pbuild.xml	1970-01-01 01:00:00.000000000 +0100
+++ util-linux-ng-2.16.1/pbuild.xml	2009-09-30 16:06:52.000000000 +0200
@@ -0,0 +1,94 @@
+<!--
+
+This file is part of the yaosp build system
+
+Copyright (c) 2009 Zoltan Kovacs
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of version 2 of the GNU General Public License
+as published by the Free Software Foundation.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+-->
+
+<build default="all">
+    <array name="files">
+        <item>fdisk/cfdisk.c</item>
+        <item>fdisk/gpt.c</item>
+        <item>fdisk/i386_sys_types.c</item>
+        <item>lib/blkdev.c</item>
+    </array>
+
+    <target name="clean">
+        <delete>objs/*</delete>
+        <rmdir>objs</rmdir>
+    </target>
+
+    <target name="prepare" type="private">
+        <mkdir>objs</mkdir>
+    </target>
+
+    <target name="compile">
+        <call target="prepare"/>
+
+        <echo>Compiling util-linux-ng 2.16.1</echo>
+
+        <for var="i" array="${files}">
+            <echo>-> ${i}</echo>
+            <gcc>
+                <input>${i}</input>
+                <output>objs/filename(${i}).o</output>
+                <include>../../../include</include>
+                <include>../../../../build/crosscompiler/lib/gcc/i686-pc-yaosp/4.3.3/include</include>
+                <include>./</include>
+                <include>./include</include>
+                <flag>-c</flag>
+                <flag>-O2</flag>
+                <flag>-m32</flag>
+                <flag>-Wall</flag>
+                <flag>-nostdinc</flag>
+                <flag>-nostdlib</flag>
+                <flag>-fno-builtin</flag>
+                <flag>-fno-strict-aliasing</flag>
+                <define key="HAVE_CONFIG_H"/>
+            </gcc>
+        </for>
+
+        <echo>Linking cfdisk</echo>
+        <echo>-> objs/cfdisk</echo>
+
+        <gcc>
+            <input>objs/*.o</input>
+            <input>../../../lib/start/objs/libstart.a</input>
+            <input>../../../thirdparty/readline-6.0/readline-6.0/objs/libreadline.a</input>
+            <input>../../../thirdparty/ncurses-5.7/ncurses-5.7/objs/libncurses.a</input>
+            <input>../../../lib/c/objs/libc.a</input>
+            <output>objs/cfdisk</output>
+            <flag>-m32</flag>
+            <flag>-Xlinker</flag>
+            <flag>--script=../../../../scripts/i386_app.lnk</flag>
+            <flag>-nostartfiles</flag>
+            <flag>-nodefaultlibs</flag>
+        </gcc>
+    </target>
+
+    <target name="install">
+        <echo>Installing cfdisk binary</echo>
+
+        <copy from="objs/cfdisk" to="../../../../build/image/application/cfdisk"/>
+    </target>
+
+    <target name="all">
+        <call target="clean"/>
+        <call target="compile"/>
+        <call target="install"/>
+    </target>
+</build>
